### PR TITLE
Fixing a problem with manual migrations

### PR DESCRIPTION
**Pull request by cgimenes**
Hi, I found a problem with manual migrations: It asks for a migration execution but it only contains some SQL instructions for locking purposes.
Looking at the logs and source code, I saw that it wasn't checking this `has-unrun-migrations?` function. So I just did some modifications to use it.

About the Exception that is thrown: `java.lang.Exception. "Database requires manual upgrade."`. I think that it should be an specific Exception, something like `ManualMigrationException`. But, I don't know how to do it, because I never worked with Clojure before, so, I kept it this way knowing it wouldn't work 
(Sorry).

During test execution, this was reported: `src/metabase/db.clj:259:5: unused-ret-vals: Var value is discarded: println` but, I don't know how to fix it either.

### CODE REVIEW

**Comment by cgimenes** [comment]
I updated the Dockerfile base image and moved one migration from data migrations to an liquibase migration, because IMO a DROP TABLE isn't a data migration.

File: src/metabase/db.clj [hunk 1]
```diff
@@ -254,6 +254,20 @@
       (catch Exception e
         (log/error e (trs "Unable to release the Liquibase lock after a migration failure"))))))
 
+(defn- print-migrations-and-quit-if-needed!
+  "If we are not doing auto migrations and there are unrun migrations then print out migration SQL for user to
+  run manually. Then throw an exception to short circuit the setup process and make it clear we can't proceed."
+  [liquibase]
+  (when (has-unrun-migrations? liquibase)
+    (println (let [sql (migrations-sql liquibase)]
```
Comment (senior) [inline src/metabase/db.clj]: I think this `println` is not correct. This was wrong in the code before as well. The SQL is being written via the log message below `log/info`. The result of that invocation will be nil, then an exception will be thrown.

Usually we always want to log things and not use `println`. In this case though, the user is starting up Metabase to see the migrations and wants that written to the console. It seems like we should swap the `log/info` for a `println`. Then we can remove this outer `println`

**Review by senior — CHANGES_REQUESTED** [review]
Regarding the `Exception`, it's very uncommon in Clojure to write your own exception subclass. It's more common to use use an `ex-info` which allows you to throw a map of data on about your error along with a message. I don't think that's needed in this case though and I think `Exception` is fine.

**Review by cgimenes — COMMENTED** [review]
Fixed

**Comment by mazameli** [comment]
Assigning this to you, @camsaul, since it deals with DB migrations. Still relevant?

**Comment by cgimenes** [comment]
.

File: resources/migrations/000_migrations.yaml [hunk 2]
```diff
@@ -5260,3 +5258,17 @@ databaseChangeLog:
                 WHERE dbcl.id = '84'
                   AND metric.updated_at < dbcl.dateexecuted
               )
+#
+# Drop the legacy QueryExecution table now that we don't need it anymore (moved from data migrations to here)
+#
+  - changeSet:
+      id: 101
+      author: camsaul
```
Comment (camsaul) [inline resources/migrations/000_migrations.yaml]: I'm not the author of this migration

**Review by camsaul — APPROVED** [review]
This LGTM, I can go ahead and fix the merge conflicts and merge this

**Comment by camsaul** [comment]
Merged in PR #12023

**Pull request closed without merging**